src/chain/chain_state.cpp

```diff
@@ -513,7 +524,13 @@ chain_state::data chain_state::to_pool(const chain_state& top,
     // If promoting from retarget height, move that timestamp into retarget.
     if (retarget &&
         is_retarget_height(height - 1u, settings.retargeting_interval))
-        data.timestamp.retarget = data.timestamp.self;
+    {
+        // litecoin fixed Satoshi's off-by-one retargeting bug.
+        if (!litecoin || height == 1)
+            data.timestamp.retarget = data.timestamp.self;
+        else
+            data.timestamp.retarget = *next(data.timestamp.ordered.crbegin());
```

Comment:
I should change this to `std::next`.

Reply:
use ternary:
```cpp
        data.timestamp.retarget = (time_warp_patch && height != 0) ? *next(data.timestamp.ordered.crbegin()) : 
            data.timestamp.self;
```

Reply:
https://github.com/libbitcoin/libbitcoin/pull/1018